File: TheBigDay/Models/Form Models/FormField.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TheBigDay.Models.Form_Models
{

    public class FormField
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid FieldId { get; set; }
        public required FieldTypes Type { get; set; }
        public required string Label { get; set; }
        public string? Placeholder { get; set; }
        public string? Description { get; set; } 
        public bool Required { get; set; }
    }

    public enum FieldTypes
    {
        STRING = 1,
        INT32,
        INT64,
        CHECKBOX,
        DROPDOWN,
        DROPDOWN_MULTISELECT,
        RICH_TEXT,
        RADIO,
        FILE_UPLOAD,
        DATE_PICKER,
        COLOR_PICKER,
        EMAIL,
    }
}

File: TheBigDay.Web/projects/common/src/lib/common-rest-models/Form/form-field.ts
export class FormField {
    fieldId?: string;
    type!: FieldType;
    label!: string;
    placeholder?: string;
    required!: boolean;
}


export enum FieldType
{
    STRING,
    INT32,
    INT64,
    CHECKBOX,
    DROPDOWN,
    DROPDOWN_MULTISELECT,
    RICH_TEXT,
    RADIO,
    FILE_UPLOAD,
    DATE_PICKER,
    COLOR_PICKER,
    EMAIL,
}

export const FieldTypeLabelRecord: Record<FieldType, string> = {
    [FieldType.STRING]: 'STRING',
    [FieldType.INT32]: 'INT32',
    [FieldType.INT64]: 'INT64',
    [FieldType.CHECKBOX]: 'CHECKBOX',
    [FieldType.DROPDOWN]: 'DROPDOWN',
    [FieldType.DROPDOWN_MULTISELECT]: 'DROPDOWN_MULTISELECT',
    [FieldType.RICH_TEXT]: 'RICH_TEXT',
    [FieldType.RADIO]: 'RADIO',
    [FieldType.FILE_UPLOAD]: 'FILE_UPLOAD',
    [FieldType.DATE_PICKER]: 'DATE_PICKER',
    [FieldType.COLOR_PICKER]: 'COLOR_PICKER',
    [FieldType.EMAIL]: 'EMAIL',
}
